### ISSUE TITLE
Pilot study: the use of an auto-merge button - this implies required status checks

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/team-compass#602]
This is following up discussion in https://github.com/jupyterhub/team-compass/issues/595#issuecomment-1340756635.

@betatim wrote:

> There was a discussion around "auto merge" in the scikit-learn repo. I learnt that once enabled a human has to activate it on a per PR basis and that there is a "override this" option (in case the branch protection rules aren't met). After learning these two things I am now in favour of activating auto-merge on repos and seeing what happens.

## Background

The auto merge button is available if you enable the following option for a protected branch.

![image](https://user-images.githubusercontent.com/3837114/206194834-3e5ee816-9f64-42ad-b533-56f14d1a2137.png)

Without specifying a specific status check, nothing will happen though!

After having specified a status check, you can observe the following:

![image](https://user-images.githubusercontent.com/3837114/206197263-4129061c-eff8-44f7-a832-c790126eaa43.png)

![image](https://user-images.githubusercontent.com/3837114/206197156-793999f7-df50-493c-ad0f-fc9876e47229.png)

**Comment by manics** [comment]
Can we enable this on a repo-by-repo basis? If the main maintainer(s) of a particular repo is happy with it are there any disadvantages to it? I see our team guidelines as a minimum that repositories must adhere to. If a repo admin enables it and later decides automerge is causing too much trouble for them they can disable it again.

**Comment by consideRatio** [comment]
> Can we enable this on a repo-by-repo basis?

Yes, yet I hope to steer us away from investing anyones time into this - I've had a bad experience with it so far.

I would like that a person going for it and activating required status checks would also be active in the repo as a maintainer with these tasks: reviewing PRs, maintaining the github workflows in general, fixing intermittent tests. I'm otherwise a vary that a person may see the benefits without the downsides that someone else may experience.

> are there any disadvantages to it?

I tried to overview those above, but here is a summary:

- Likeley to run more github jobs than needed, wasting compute resources and time to green status for documentation only changes etc
- Likeley to add more complexity in workflows if trying to address
- Likeley to require repo admins to update branch protection rules if status checks changes, and manage such complexity
- Likeley that we choose to escalate maintainer status from repo write -> repo admin just because we want to provide the conveniece for all with merge permissions to bypass the required status checks, which is often relevant because something may be intermittent or known to be an unrelated failure
